### PR TITLE
Add labeler action

### PR DESCRIPTION
**Pull request by nlohmann**
Automatically adds labels to PRs:

- size label (S, M, L) according to the number of additions+deletions
- other labels:
  - documentation
  - CI
  - CMake
  - tests

The idea is to make it easier to grasp what is touched by a CI, and make also to see on first sight that a PR without tests could be an issue.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -9,6 +9,18 @@ labels:
...
+  files:
+    - "tests/.*"
+
+- label: "CMake"
+  files:
+    - ".*CMakeLists.txt"
+
+- label: "CMake"
+  files:
+    - "cmake/.*"
```
Comment (falbrechtskirchinger) [inline .github/labeler.yml:22]: No need to add the label twice?
```suggestion
- label: "CMake"
  files:
    - ".*CMakeLists.txt"
    - "cmake/.*"
```

**Review by falbrechtskirchinger — CHANGES_REQUESTED** [review]

File: .github/labeler.yml [hunk 2]
```diff
@@ -21,11 +21,11 @@ labels:
   files:
     - "cmake/.*"
 
-- label: "ci"
+- label: "CI"
   files:
     - "github/workflows/.*"
 
-- label: "ci"
+- label: "CI"
```
Comment (falbrechtskirchinger) [inline .github/labeler.yml:28]: same here.

File: .github/labeler.yml (hunk 1 above)
Comment (nlohmann) [reply .github/labeler.yml:22]: No, the files must match *all* listed regular expressions, so it's an implicit conjunction. For disjunctions, we need to repeat the rule.

File: .github/labeler.yml (hunk 2 above)
Comment (nlohmann) [reply .github/labeler.yml:28]: See above.

File: .github/labeler.yml (hunk 1 above)
Comment (falbrechtskirchinger) [reply .github/labeler.yml:22]: Wait, this is not the official GitHub labeler? I'm looking at the wrong documentation.

Comment (nlohmann) [reply .github/labeler.yml:22]: See https://github.com/srvaroa/labeler#configuration:

> The action combines all conditions with an AND operation.
> ...
> If you wish to apply an OR, you may set multiple matchers for the same label.

Comment (nlohmann) [reply .github/labeler.yml:22]: Right, it's not the official one. This one has more features and is actively developed.

Comment (falbrechtskirchinger) [reply .github/labeler.yml:22]: I'm looking at:
`- uses: actions/labeler@v4`
https://github.com/marketplace/actions/labeler

But you're using:
`- uses: srvaroa/labeler@master`

**Review by falbrechtskirchinger — APPROVED** [review]
Looks good to me.

**Comment by Nhattan9961** [comment]
S

**Pull request merged**